engine/src/main/java/org/terasology/logic/characters/CharacterSystem.java

```diff
@@ -73,10 +74,21 @@
     @In
     private Time time;
 
+    @In
+    private TranslationSystem translationSystem;
+
     @ReceiveEvent(components = {CharacterComponent.class})
     public void onDeath(DoDestroyEvent event, EntityRef entity) {
         CharacterComponent character = entity.getComponent(CharacterComponent.class);
-        character.controller.send(new DeathEvent());
+        DeathEvent deathEvent = new DeathEvent();
+        //Store the details of the death in the event for display on the death screen
+        deathEvent.lastDamageType = event.getDamageTypeString();
```

Comment:
rename both, method and field to `damageTypeName`, `getDamageTypeName`. Same for the instigator.